packages/wallet/src/utils/transaction-generator.ts

```diff
@@ -1,14 +1,26 @@
 import {TransactionRequest} from "ethers/providers";
-import {getAdjudicatorInterface} from "./contract-utils";
-import {splitSignature} from "ethers/utils";
+import {getAdjudicatorInterface, getAdjudicatorContractAddress} from "./contract-utils";
+import {splitSignature, BigNumber} from "ethers/utils";
 import {Commitment, SignedCommitment, signCommitment2} from "../domain";
 import {asEthersObject} from "fmg-core";
 import {
   createDepositTransaction as createNitroDepositTransaction,
+  createTransferAllTransaction as createNitroTransferAllTransaction,
   Transactions as nitroTrans,
   SignedState
 } from "@statechannels/nitro-protocol";
-import {convertCommitmentToState} from "./nitro-converter";
+import {convertAddressToBytes32, convertCommitmentToState} from "./nitro-converter";
+import {Allocation, AllocationItem} from "@statechannels/nitro-protocol/src/contract/outcome";
+
+interface GetDataResult {
+  turnNumRecord: number;
+  finalizesAt: number;
+  fingerprint: BigNumber;
+}
+
+export function nitroGetData(provider, channelId: string): Promise<GetDataResult> {
+  return nitroTrans.getData(provider, getAdjudicatorContractAddress(), channelId);
+}
```

Comment:
This seems like a helper function that does not need to exist

Reply:
Yeah it does not  - will remove.